lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt

```diff
@@ -80,52 +84,142 @@ import kotlinx.serialization.json.*
 @Suppress("DataClassPrivateConstructor")
 @Serializable
 data class TypedData private constructor(
-    val types: Map<String, List<TypeBase>>,
+    @SerialName("types")
+    val customTypes: Map<String, List<Type>>,
+
```

Comment:
I don't have a strong opinion about this, so whatever you think is best. I generally like the idea of keeping the naming as close to json as possible though. it may have been a bit less descriptive at first, but it follows closer the json which is a big plus imo.